src/functionlenses.jl

```diff
@@ -42,6 +42,12 @@ function set(obj::NamedTuple, ::Type{NamedTuple{KS}}, val::NamedTuple) where {KS
     setproperties(obj, NamedTuple{KS}(val))
 end
 
+function set(obj, f::Base.Fix1{typeof(getindex)}, val)
+    ix = findfirst(isequal(val), f.x)
+    ix === nothing && throw(ArgumentError("value $val not found in $(f.x)"))
+    return ix
+end
```

Comment:
I have some objections. 
* `findfirst` is typically `O(n)`, preferably optics are fast 
* If there are multiple matches, I would prefer an error.